### PR TITLE
X3d

### PR DESCRIPTION
**Pull request by mwitchwilliams**
X3D demo files.  Parser code starts the animations.

GearVRf-DCO-1.0-Signed-off-by: Mitch Williams 
m1.williams@partner.samsung.com

### CODE REVIEW

File: gvr-widgetviewer/app/src/main/.classpath [hunk 1]
```diff
@@ -4,6 +4,7 @@
 	<classpathentry kind="con" path="com.android.ide.eclipse.adt.ANDROID_FRAMEWORK"/>
 	<classpathentry exported="true" kind="con" path="com.android.ide.eclipse.adt.LIBRARIES"/>
 	<classpathentry exported="true" kind="con" path="com.android.ide.eclipse.adt.DEPENDENCIES"/>
+	<classpathentry kind="src" path="src"/>
```
Comment (rahul27) [inline gvr-widgetviewer/app/src/main/.classpath]: This file need not be checked in.

File: gvr-x3d-demo/gvr-x3d-demo/build.gradle [hunk 2]
```diff
@@ -0,0 +1,28 @@
+apply plugin: 'com.android.application'
```
Comment (rahul27) [inline gvr-x3d-demo/gvr-x3d-demo/build.gradle]: to maintain consistency with the other apps the folder structure needs to be **gvr-x3d-demo/app**

File: gvr-x3d-demo/gvr-x3d-demo/src/main/java/org/gearvrf/x3ddemo/X3DparserActivity.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    @Override
+    protected void onCreate(Bundle icicle) {
+        super.onCreate(icicle);
+        setScript(script, "gvr_note4.xml");
+        
+        mWebView = new GVRWebView( this );
+    }
+  
+  public GVRWebView getWebView() {
+      return mWebView;
```
Comment (rahul27) [inline gvr-x3d-demo/gvr-x3d-demo/src/main/java/org/gearvrf/x3ddemo/X3DparserActivity.java]: GVRWebView is not required, you can get rid of it.

File: gvr-x3d-demo/settings.gradle [hunk 4]
```diff
@@ -0,0 +1,3 @@
+include ':framework'
+include ':gvrsample'
```
Comment (rahul27) [inline gvr-x3d-demo/settings.gradle]: needs to be include ':app'

**Comment by rahul27** [comment]
The app crashes on launch:

````
07-18 16:20:52.431 588-710/com.samsung.android.gvr.sample D/AsyncBitmapTexture: Actual GL_MAX_TEXTURE_SIZE = 8192
07-18 16:20:52.431 588-710/com.samsung.android.gvr.sample D/GVRAssetLoader: ASSET: loading teapotandtorus.x3d ...
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample E/GVRAssetLoader: ASSET: ERROR: model teapotandtorus.x3d did not load X3D not supported yet
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err: java.lang.NullPointerException: Attempt to invoke virtual method 'java.util.List org.gearvrf.scene_objects.GVRModelSceneObject.getAnimations()' on a null object reference
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.x3ddemo.X3DparserScript.onInit(X3DparserScript.java:116)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at java.lang.reflect.Method.invoke(Native Method)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at java.lang.reflect.Method.invoke(Method.java:372)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVREventManager.invokeMethod(GVREventManager.java:307)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVREventManager.sendEventWithMaskParamArray(GVREventManager.java:126)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVREventManager.sendEventWithMask(GVREventManager.java:113)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVREventManager.sendEvent(GVREventManager.java:108)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVRViewManager$3$1.run(GVRViewManager.java:695)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVRContext.runOnTheFrameworkThread(GVRContext.java:2779)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVRViewManager$3.beforeDrawEyes(GVRViewManager.java:687)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVRViewManager.beforeDrawEyes(GVRViewManager.java:342)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.GVRActivity$1.onBeforeDrawEyes(GVRActivity.java:136)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at org.gearvrf.VrapiActivityHandler$7.onDrawFrame(VrapiActivityHandler.java:410)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1561)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample W/System.err:     at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1278)
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample D/SplashScreen: close()
07-18 16:20:52.571 588-710/com.samsung.android.gvr.sample W/VrApi: Time to first swap: 0.972 seconds
````

**Comment by rahul27** [comment]
Also you might want to squash all the commits.

**Comment by liaxim** [comment]
Yes, squashing the commits would be greatly appreciated. There is really no good reason to have more than one commit.

File: gvr-x3d-demo/gvr-x3d-demo/src/main/java/org/gearvrf/x3ddemo/X3DparserActivity.java (hunk 3 above)
Comment (mwitchwilliams) [inline gvr-x3d-demo/gvr-x3d-demo/src/main/java/org/gearvrf/x3ddemo/X3DparserActivity.java]: done.
Note that X3D (being a web based technology) can open web pages, but currently that feature isn't supported.

**Comment by mwitchwilliams** [comment]
Regarding the crash error message:
ASSET: loading teapotandtorus.x3d ...
07-18 16:20:52.441 588-710/com.samsung.android.gvr.sample E/GVRAssetLoader: ASSET: ERROR: model teapotandtorus.x3d did not load X3D not supported yet
This was tested with an older version of GVRAssetLoader since the response "X3D not supported yet" no longer exists there.

File: gvr-widgetviewer/app/src/main/.classpath (hunk 1 above)
Comment (mwitchwilliams) [inline gvr-widgetviewer/app/src/main/.classpath]: Not sure how I changed that, but that commit would be my mistake.  Is there a way to back out this change?

**Comment by liaxim** [comment]
Bunch of commits in the X3D pull request for master that we merged yesterday are missing compared to the pull request we closed.

**Comment by rahul27** [comment]
In that case you might want to remove that file from this sample and add it back when it has been fixed.

**Comment by liaxim** [comment]
Nola and Mitch need to diff the pull request we merged and the one we closed. I can't just cherry-pick the missing commits - there are various conflicts that would be best resolved by Nola and Mitch.

**Comment by rahul27** [comment]
replaced by https://github.com/gearvrf/GearVRf-Demos/pull/262

**Pull request closed without merging**